// Copyright (c) 2023 Boston Dynamics, Inc.  All rights reserved.
//
// Downloading, reproducing, distributing or otherwise using the SDK Software
// is subject to the terms and conditions of the Boston Dynamics Software
// Development Kit License (20191101-BDSDK-SL).

syntax = "proto3";

package bosdyn.api;
option go_package = "bosdyn/api/fault_service";

option java_outer_classname = "FaultServiceProto";

import "bosdyn/api/service_fault.proto";

// The service fault service enables modification of the robot state ServiceFaultState.
service FaultService {
    // Sends a ServiceFault to be reporting in robot state.
    rpc TriggerServiceFault(TriggerServiceFaultRequest) returns (TriggerServiceFaultResponse) {}

    // Clears an active ServiceFault from robot state.
    rpc ClearServiceFault(ClearServiceFaultRequest) returns (ClearServiceFaultResponse) {}
}
